Geometry/CaloGeometry/interface/CaloCellGeometry.h

```diff
@@ -76,15 +76,18 @@ class CaloCellGeometry
   virtual ~CaloCellGeometry() ;
       
   /// Returns the corner points of this cell's volume.
-  CornersVec const & getCorners() const { assert(not m_corners.uninitialized()); return m_corners; }
-  RepCorners const & getCornersREP() const {  return m_repCorners;}
+  CornersVec getCorners() const { assert(not m_corners.uninitialized()); return m_corners; }
+  RepCorners getCornersREP() const {  return m_repCorners;}
```

Comment:
why were these made to return a copy?
The lifetime of m_corners should be the same as the CaloCellGeometry itself.

Same question about the positions below.

Reply:
I asked Chris and he said it is safer to send by copy rather than by reference and once this is done in this class the same holds for PFRecHit

Reply:
I shall make changes only when you get things clarified for me

Reply:
This would be fine to return by `const &`. The problem was functions which were like
```
Foo const& getFoo() const { return functionThatReturnsANewlyMadeSharedPtr()->foo(); }
```
In such a case the `std::shared_ptr<>` returned from `functionThatReturnsANewlyMadeSharedPtr()` is deleted just as the function ends and therfore the `Foo const&` returned points to invalid memory.